src/job/files/idx.rs

```diff
@@ -48,61 +55,84 @@ pub(crate) async fn write_parquet_index_to_disk(
     stream_type: StreamType,
     file_name: &str,
     caller: &str,
-) -> Result<(String, FileMeta, StreamType), anyhow::Error> {
+) -> Result<Vec<(String, FileMeta)>, anyhow::Error> {
     let schema = if let Some(first_batch) = batches.first() {
         first_batch.schema()
     } else {
         return Err(anyhow::anyhow!("No record batches found"));
     };
+    let partition = StreamPartition::new_prefix("term");
 
-    // write metadata
-    let mut file_meta = FileMeta {
-        min_ts: 0,
-        max_ts: 0,
-        records: 0,
-        original_size: file_size as i64,
-        compressed_size: 0,
-        flattened: false,
-    };
-    populate_file_meta(
-        schema.clone(),
-        vec![batches.to_vec()],
-        &mut file_meta,
-        Some("min_ts"),
-        Some("max_ts"),
-    )
-    .await?;
+    let json_rows = record_batches_to_json_rows(&batches.iter().collect::<Vec<_>>())?;
 
-    // write parquet file
-    let mut buf_parquet = Vec::new();
-    let mut writer = new_parquet_writer(&mut buf_parquet, &schema, &[], &file_meta);
-    for batch in batches {
-        writer.write(&batch).await?;
+    let mut partition_buf: HashMap<String, Vec<Arc<serde_json::Value>>> = HashMap::new();
+    for row in json_rows {
+        let val = match row.get("term") {
+            Some(v) => get_string_value(v),
+            None => "null".to_string(),
+        };
+        let prefix = partition.get_partition_key(&val);
+        let batch = partition_buf.entry(prefix).or_insert_with(|| Vec::new());
+        let entry = serde_json::Value::Object(row);
+        batch.push(Arc::new(entry));
     }
-    writer.close().await?;
-    file_meta.compressed_size = buf_parquet.len() as i64;
+    let mut ret = Vec::new();
 
-    let new_idx_file_name =
-        generate_index_file_name_from_compacted_file(org_id, stream_type, stream_name, file_name);
-    log::info!(
-        "[JOB] IDX: write_to_disk: {}/{}/{} {} {} {}",
-        org_id,
-        stream_name,
-        stream_type,
-        new_idx_file_name,
-        file_name,
-        caller,
-    );
+    for (prefix, records) in partition_buf.into_iter() {
+        // write metadata
+        let mut file_meta = FileMeta {
+            min_ts: 0,
+            max_ts: 0,
+            records: 0,
+            original_size: file_size as i64,
+            compressed_size: 0,
+            flattened: false,
+        };
+        let batch = convert_json_to_record_batch(&schema, &records)?;
+        populate_file_meta(
+            schema.clone(),
```

Comment:
I'm not sure if there is any better way to do this. After compaction we have a new list of records for new index file, and to split it into multiple index files based on prefix partition , I need to convert them to json, get `term` and split those json into partitions, and finally convert them back into record batches.